### PR TITLE
fix(stdlib): Fixed String.writeUtf8CodePoint for two byte sequences.

### PR DESCRIPTION
**Pull request by cician**
The two bit prefix was applied incorrectly on the second byte. Also, changed the format of bit masking and prefixing constants to binary for easier reasoning about the encoding.

### CODE REVIEW

File: stdlib/string.gr [hunk 1]
```diff
@@ -975,27 +975,44 @@ let writeUtf8CodePoint = (ptr, codePoint) => {
...
     1n
   } else if (codePoint <= 0x07FFn) {
-    let high = ((codePoint >>> 6n) & 0x1Fn) | 0xC0n
-    let low = (codePoint & 0x3Fn) | 0x08n
+    // Code points in the range 0x0080..0x07FF are written as two bytes.
+    // The first byte has a three bit prefix of 110, followed by 5 bits of the
+    // codepoint. The second byte has a two bit prefix of 10, followed by 6 bits
+    // of the codepoint.
+		let high = ((codePoint >>> 6n) & 0b000_11111n) | 0b110_00000n
+		let low =  ( codePoint         & 0b00_111111n) | 0b10_000000n
```
Comment (peblair) [inline stdlib/string.gr:991]: @ Other reviewers: This appears to be the changed line (there was a typo in the original: `0x08n` should have been `0x80n`)....

**Review by peblair — COMMENTED** [review]
@cician Apologies for the delay; this looks great! One question: is there a test case we could add to capture the bug which was fixed?

**Comment by cician** [comment]
No problem for the delay. I'm on vacation myself and AFK until september.

Maybe the simplest test would be to encode and decode back a string with a codepoint of each length. I can write one when I'm back home.

**Comment by phated** [comment]
@cician hope you had a great vacation! I'd love to get this landed for our 0.4 release (hopefully will happen this upcoming week). Will you have a chance to add regression tests by then? Cheers 🍻

**Review by peblair — APPROVED** [review]
I went ahead and threw in a simple test (`"¢"` is encoded as `[0xc2, 0xa2]`)...I cheated a little because Grain natively uses UTF8 🙂 

@cician Thanks for the patch!

File: stdlib/string.gr (hunk 1 above)
Comment (phated) [inline stdlib/string.gr:991]:
```suggestion
    // Code points in the range 0x0080..0x07FF are written as two bytes.
    // The first byte has a three bit prefix of 110, followed by 5 bits of the
    // codepoint. The second byte has a two bit prefix of 10, followed by 6 bits
    // of the codepoint.
    let high = ((codePoint >>> 6n) & 0b000_11111n) | 0b110_00000n
    let low =  ( codePoint         & 0b00_111111n) | 0b10_000000n
```
I think the indentation was just a case of tabs v spaces. (Also, I'm going to commit these so I can see if the suite passes).

File: stdlib/string.gr [hunk 2]
```diff
@@ -975,27 +975,44 @@ let writeUtf8CodePoint = (ptr, codePoint) => {
...
-    let high = ((codePoint >>> 12n) & 0x0Fn) | 0xE0n
-    let mid = ((codePoint >>> 6n) & 0x3Fn) | 0x80n
-    let low = (codePoint & 0x3Fn) | 0x80n
+    // Code points in the range 0x0800..0xFFFF are written as three bytes.
+    // The first byte has a four bit prefix of 1110, followed by 4 bits of the
+    // codepoint. Remaining bytes each have a two bit prefix of 10, followed by
+    // 6 bits of the codepoint.
+    let high = ((codePoint >>> 12n) & 0b0000_1111n) | 0b1110_0000n
+		let mid =  ((codePoint >>> 6n)  & 0b00_111111n) | 0b10_000000n
+		let low =  ( codePoint          & 0b00_111111n) | 0b10_000000n
```
Comment (phated) [inline stdlib/string.gr:1002]:
```suggestion
    // Code points in the range 0x0800..0xFFFF are written as three bytes.
    // The first byte has a four bit prefix of 1110, followed by 4 bits of the
    // codepoint. Remaining bytes each have a two bit prefix of 10, followed by
    // 6 bits of the codepoint.
    let high = ((codePoint >>> 12n) & 0b0000_1111n) | 0b1110_0000n
    let mid =  ((codePoint >>> 6n)  & 0b00_111111n) | 0b10_000000n
    let low =  ( codePoint          & 0b00_111111n) | 0b10_000000n
```
grain format when 😉

File: stdlib/string.gr [hunk 3]
```diff
@@ -975,27 +975,44 @@ let writeUtf8CodePoint = (ptr, codePoint) => {
...
-    let mid2 = ((codePoint >>> 6n) & 0x3Fn) | 0x80n
-    let low = (codePoint & 0x3Fn) | 0x80n
+    // Code points in the range 0x10000..0x10FFFF are written as four bytes.
+    // The first byte has a five bit prefix of 11110, followed by 3 bits of the
+    // codepoint. Remaining bytes each have a two bit prefix of 10, followed by
+    // 6 bits of the codepoint.
+    let high = ((codePoint >>> 18n) & 0b00000_111n) | 0b11110_000n
+		let mid1 = ((codePoint >>> 12n) & 0b00_111111n) | 0b10_000000n
+		let mid2 = ((codePoint >>> 6n)  & 0b00_111111n) | 0b10_000000n
+		let low =  ( codePoint          & 0b00_111111n) | 0b10_000000n
```
Comment (phated) [inline stdlib/string.gr:1015]:
```suggestion
    // Code points in the range 0x10000..0x10FFFF are written as four bytes.
    // The first byte has a five bit prefix of 11110, followed by 3 bits of the
    // codepoint. Remaining bytes each have a two bit prefix of 10, followed by
    // 6 bits of the codepoint.
    let high = ((codePoint >>> 18n) & 0b00000_111n) | 0b11110_000n
    let mid1 = ((codePoint >>> 12n) & 0b00_111111n) | 0b10_000000n
    let mid2 = ((codePoint >>> 6n)  & 0b00_111111n) | 0b10_000000n
    let low =  ( codePoint          & 0b00_111111n) | 0b10_000000n
```
last one

**Review by phated — APPROVED** [review]
Let's go! Thanks for this PR @cician 🔥

**Pull request merged**